### ISSUE TITLE
Variable declarations are missing

### ISSUE THREAD

**Issue by falsandtru** [Microsoft/TypeScript#6171]
```ts
// a.ts
var A: void;
export {A}

// types.ts
import {A} from './a';
export {A}
declare type A = void; // this code remove a variable declaration.

// index.ts
import {A} from './types';
var B = A;
var C: A;
```

```
$ node built/local/tsc.js -m commonjs index --noEmit
index.ts(2,9): error TS2304: Cannot find name 'A'.
```

Remove a type declaration code.

```diff
- declare type A = void;
```

Work correctly.

```
$ node built/local/tsc.js -m commonjs index --noEmit
index.ts(3,8): error TS2304: Cannot find name 'A'.
```

**Comment by ahejlsberg** [comment]
This is by design. When resolving an alias we stop as soon as we find a non-alias. In this case that means we stop resolution on type A, and we won't see the further removed declaration of variable A. Effectively, all meanings of a symbol must come from the same file. The documentation is not entirely clear on this, and we already have a bug to fix that (#3158).

**Comment by falsandtru** [comment]
I understand. I want to read these type resolution strategies and type system policy for my bug reporting. Thanks for your explain.
